### PR TITLE
TFIFFHIR-428 

### PR DESCRIPTION
**Pull request by bnorris2**
TFIFFHIR-428 change to search dependencies for base definition snapshot

### CODE REVIEW

File: apps/server/src/app/fhir.controller.ts
Comment (seanmcilvenna) [inline apps/server/src/app/fhir.controller.ts:698]: I think you can do `@Query('isLightweight') isLightweight = true` to set it as a default, and then later you can always check for `if (isLightweight) { ... }`

File: apps/server/src/app/structure-definition.controller.ts
Comment (seanmcilvenna) [inline apps/server/src/app/structure-definition.controller.ts:117]: because `return ret;` is in the line above, I don't think you need the `} else { ... }`
you can just call the code _in_ the `else` after the `if (dependencies.entry.length === 1) { ... }` block because if there _is_  a single dependency, then it will stop the code from progressing
the code works either way, it is just my preference not to use `else` when it is not needed...

**Review by seanmcilvenna — CHANGES_REQUESTED** [review]

**Comment by seanmcilvenna** [comment]
After you review these comments and make whatever changes you feel are appropriate, I'm going to hold off on merging this until after I merge `development` into the pre-release; it could be a couple days before this gets merged, even after the review/comments are addressed.

File: apps/server/src/app/fhir.controller.ts
Comment (bnorris2) [reply apps/server/src/app/fhir.controller.ts:698]: ah! good catch, thanks

File: apps/server/src/app/structure-definition.controller.ts
Comment (bnorris2) [reply apps/server/src/app/structure-definition.controller.ts:117]: very true, good point, thank you

**Comment by bnorris2** [comment]
I've reviewed/made the changes. Feel free to push whenever you merge the other branch in, thanks

**Pull request merged**
